ibm/service/power/resource_ibm_pi_network_interface.go

```diff
@@ -0,0 +1,356 @@
+
+	"github.com/IBM-Cloud/power-go-client/power/models"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/conns"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/flex"
+)
+
+func ResourceIBMPINetworkInterface() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceIBMPINetworkInterfaceCreate,
+		ReadContext:   resourceIBMPINetworkInterfaceRead,
+		UpdateContext: resourceIBMPINetworkInterfaceUpdate,
+		DeleteContext: resourceIBMPINetworkInterfaceDelete,
+		Importer:      &schema.ResourceImporter{},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(60 * time.Minute),
+			Delete: schema.DefaultTimeout(60 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			// Arguments
+			Arg_CloudInstanceID: {
+				Description:  "The GUID of the service instance associated with an account.",
+				ForceNew:     true,
+				Required:     true,
+				Type:         schema.TypeString,
+				ValidateFunc: validation.NoZeroValues,
+			},
+			Arg_InstanceID: {
+				Description: "If supplied populated it attaches to the InstanceID, if empty detaches from InstanceID.",
+				Optional:    true,
+				Type:        schema.TypeString,
+			},
+			Arg_IPAddress: {
+				Description: "The requested IP address of this Network Interface.",
+				ForceNew:    true,
+				Optional:    true,
+				Type:        schema.TypeString,
+			},
+			Arg_Name: {
+				Description: "Name of the Network Interface.",
+				Optional:    true,
+				Type:        schema.TypeString,
+			},
+			Arg_NetworkID: {
+				Description:  "Network ID.",
+				ForceNew:     true,
+				Required:     true,
+				Type:         schema.TypeString,
+				ValidateFunc: validation.NoZeroValues,
+			},
+			Arg_UserTags: {
+				Description: "The user tags attached to this resource.",
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Optional:    true,
+				Set:         schema.HashString,
+				Type:        schema.TypeSet,
+			},
+			// Attributes
+			Attr_CRN: {
+				Computed:    true,
+				Description: "The Network Interface's crn.",
+				Type:        schema.TypeString,
+			},
+			Attr_Instance: {
+				Computed:    true,
+				Optional:    true,
+				Description: "The attached instance to this Network Interface.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						Attr_Href: {
+							Computed:    true,
+							Description: "Link to instance resource.",
+							Type:        schema.TypeString,
+						},
+						Attr_InstanceID: {
+							Computed:    true,
+							Description: "The attached instance ID.",
+							Type:        schema.TypeString,
+						},
+					},
+				},
+				Type: schema.TypeList,
+			},
+			Attr_IPAddress: {
+				Computed:    true,
+				Description: "The ip address of this Network Interface.",
+				Type:        schema.TypeString,
+			},
+			Attr_MacAddress: {
+				Computed:    true,
+				Description: "The mac address of the Network Interface.",
+				Type:        schema.TypeString,
+			},
+			Attr_Name: {
```

Comment:
There is already `Arg_Name`.

Reply:
@yussufsh If the user does not provide a name during creation, one will be automatically generated which creates diff if we were to to another apply after creation. Hence this we have an attribute to avoid that.